docs/operation.md

```diff
@@ -0,0 +1,624 @@
+# Working with Epic Connector
+[[  Overview ]](#overview)  [[ Operation details ]](#operation-details)  [[  Sample configuration  ]](#sample-configuration)
+
+### Overview
+The following operations allow you to work with buckets in Epic. Click an operation name to see details on how to use it.
+For a sample proxy service that illustrates how to work with buckets, see [Sample configuration](#sample-configuration).
+
+| Operation        | Description |
+| ------------- |-------------|
+| [readById](#retrieve-a-specific-resource-by-id)| Retrieve a specific resource by its Id. |
+| [searchAllergyIntolerance](#search-allergy-intolerance)| Search clinical information about a patient's allergic response to a substance. |
+| [searchCarePlan](#search-care-plan)| Search the plan of treatment for a patient. |
+| [searchCondition](#search-condition)| Search conditional state of the patient. |
+| [searchDevice](#search-device)| Search medical deivces implanted in a patient. |
+| [searchDiagnosticReport](#search-diagnostic-report)| Search information about a single diagnostic service performed for a patient. |
+| [searchDocumentReference](#search-document-reference)| Search information about a document. |
+| [searchFamilyMemberHistory](#search-family-member-history)| Search a patient's family history. |
+| [searchGoal](#search-goal)| Search objectives for a patient based on a current condition. |
+| [searchImmunization](#search-immunization)| Search information about a patient's immunizations. |
+| [searchMedicationOrder](#search-medication-order)| Search information about a medication that was ordered for a patient. |
+| [searchMedicationStatement](#search-medication-statement)| Search information about medications being consumed by a patient. |
+| [searchObservation](#search-observation)| Search  information about a measurement or simple assertion made about a patient.  |
+| [searchPatient](#search-patient)| Search Patient using the given parameters. |
+| [searchProcedure](#search-procedure)| Search information about a measurement or simple assertion made about a patient |
+
+
+## Operation details
+This section provides details on each of the operations.
+
+## Retrieve a Specific Resource by Id
+This readById operation reads a resource based on the Id you provide.
+
+**ReadById**
+```xml
+<Epic.readById>
+    <base>{$ctx:base}</base>
+    <type>{$ctx:type}</type>
+    <id>{$ctx:id}</id>
+</Epic.readById>
+```
+**Properties**
+* base: The service root URL.
+* type: The type of the resource that is needed to be retrieved.
+* id: The logical Id of the resource.
+
+**Sample request**
+
+Following is a sample REST request that can be handled by the readById operation.
+```
+{
+ "base": “https://open-ic.epic.com/FHIR/api/FHIR/DSTU2/",
+ "type": “Binary",
+ "id": "T51iWjMg3MhfQWNRaJpmEF8ALTC6lQcunX-VTNpxqyTiV8.6oSleCNUa5Yr82YE4CJaCO0rcZankXUhpoCWkzlwB"
+}
+```
```

Comment:
Add sample response for each operations as in [1].
[1] https://docs.wso2.com/display/ESBCONNECTORS/Working+with+Sheet+Operations